C++/Binary_Search.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+using namespace std;
+int binarysearch(int *a, int n, int x)
+{
+//  start variable denotes the starting point in an array.
+//  end variable denotes the end point of array.
+//  mid variable to find the mid value between them.
+    int start = 0;
+    int end = n - 1;
+    int mid;
+    while (end >= start)
+    {
+        mid = (start + end) / 2;
+        if (a[mid] == x)
+        {
+            return mid;
+        }
+        else if (a[mid] > x)
+        {
+            end = mid - 1;
+        }
+        else
+        {
+            start = mid + 1;
+        }
+    }
+    return -1;
+}
+int main()
+{
+    // Input the size of the array
```

Comment:
you can add here like cout << "Enter size of array";
so user who running the program can know what type of input you are asking for.